lib/internal/Magento/Framework/Image/Adapter/AbstractAdapter.php

```diff
@@ -675,7 +675,10 @@ protected function _prepareDestination($destination = null, $newName = null)
             try {
                 $this->directoryWrite->create($this->directoryWrite->getRelativePath($destination));
             } catch (\Magento\Framework\Filesystem\FilesystemException $e) {
-                $this->logger->critical($e);
+                if (null !== $this->logger) {
```

Comment:
I suspect the bug is the logger property is not being initialized in constructor. Add it to constructor list and DI framework will provide it.